### ISSUE TITLE
Foxy Version Fails with Cmake 3.10 - Unknown CMake command "target_link_directories".'

### ISSUE THREAD

**Issue by mrawding** [ros2/libyaml_vendor#33]
In the master version, it does a check for the CMake version before using the target_link_directories command. However, on the foxy branch it does not. This results in an error when building ROS foxy from source with a CMake version less than 3.13, although the minimum required is 3.9.

This is in the master version and in foxy-backport. Which should work if ros_base foxy repos pointed to master.

```    if(${CMAKE_VERSION} VERSION_LESS "3.13")
      link_directories(${yaml_LIBRARY_DIRS})
    else()
      target_link_directories(test_yaml_reader PRIVATE ${yaml_LIBRARY_DIRS})
```

However it seems that ros_base links to the foxy library and not foxy-backport. Is there a way we can change this for the source install from r
```
 rosinstall_generator ros_base --deps --rosdistro foxy > ros2-base.repos                                                                                                                                    

``` 

So that it points to the correct branch?

**Comment by clalancette** [comment]
See #29.  Once that is merged, the problem will be fixed on the 'foxy' branch.

**Comment by clalancette** [comment]
This was fixed in #29, so closing.
